src/lib.rs

```diff
@@ -39,19 +39,26 @@ impl TryFrom<zvariant::Value<'_>> for SchemePreference {
 impl<'s> TryFrom<&'s zbus::Message> for SchemePreference {
     type Error = Error;
     fn try_from(message: &'s zbus::Message) -> Result<Self> {
-        message
-            .body::<(&str, &str, zvariant::Value<'_>)>()
-            .and_then(|args| SchemePreference::try_from(args.2))
+        let body = message.body();
+        let value: zvariant::Value = body.deserialize()?;
+        SchemePreference::try_from(value)
```

Comment:
```suggestion
        let args: (&str, &str, zvariant::Value<'_>) = body.deserialize()?;
        SchemePreference::try_from(args.2)
```

This fixes the Issue for me